lib/resque/data_store.rb

```diff
@@ -317,16 +317,16 @@ def stat(stat)
         @redis.get("stat:#{stat}").to_i
       end
 
-      def increment_stat(stat, by = 1)
-        @redis.incrby("stat:#{stat}", by)
+      def increment_stat(stat, by = 1, redis: @redis)
```

Comment:
non blocking, but using an instance variable as the default value for a param feels a little too fancy. Adding `redis ||= @redis` in the method body feels more straightforward and easy to read.

Reply:
Hmm... I hadn't thought of it that way.  I suppose that's fair.

I've usually preferred to put optional & kwarg defaults into the params list because I've assumed that putting defaults there will pass through into rdoc/yard whereas defaults inside the body won't.  But I have to admit, it's been a little while since I've verified that that's true, so I might be misremembering.  Doing it this way *should* update rdoc/yard, IMO.  😉  Even if it doesn't, it feels a bit more self-documenting to me.

But of course, YMMV!  Thanks for the feedback.  I'll be pondering it for a little while. 🙂